kubernetes/process-manifest.rb

```diff
@@ -0,0 +1,86 @@
+require "yaml"
+require "fileutils"
+
+if ARGV.count != 3
+  puts "usage: #{$0} container-manifests-dir velum-source-code-dir salt-states-dir"
+  Process.exit 1
+end
+
+def container_volume(name:, path:)
+  {
+    "name" => volume_name(name: name),
+    "mountPath" => path
+  }
+end
+
+def patch_containers(yaml)
+  yaml["spec"]["volumes"] ||= []
+  yaml["spec"]["containers"].each do |container|
+    if container["image"] =~ /^sles12/
+      container["image"] = "docker-testing-registry.suse.de/#{container["image"]}"
+    end
+    container["env"] ||= []
+    container["env"].each do |envvar|
+      envvar["value"] = "development" if envvar["name"] == "RAILS_ENV"
+    end
+    container["volumeMounts"] ||= []
+    container["volumeMounts"].reject! do |volume_mount|
+      ["/srv/salt", "/srv/pillar", "/srv/velum"].include? volume_mount["mountPath"]
+    end
+    container["volumeMounts"] += case container["name"]
+    when "salt-master"
+      [
+        container_volume(name: "salt-states", path: "/srv/salt"),
+        container_volume(name: "salt-pillar", path: "/srv/pillar")
+      ]
+    when "velum-dashboard", "velum-event-processor"
+      [
+        container_volume(name: "velum-source-code", path: "/srv/velum")
+      ]
+    else
+      []
+    end
+  end
+end
+
+def volume_name(name:)
+  "dev-env-#{name}"
```

Comment:
Yes, but it's called from two different places, and those two should always match. If you change one and forget the other the manifest won't be valid. This way we ensure that if you rename it, you are forcefully renaming both.